python/packages/ai/teams/app.py

```diff
@@ -211,7 +211,7 @@ async def on_event(context: TurnContext, state: TurnState):
         """
 
         def __selector__(context: TurnContext):
-            return type == str(context.activity.type)
```

Comment:
we need this cast because the bot framework type for `context.activity.type` is `Unknown | None`, because they did not add type annotations.

Reply:
Ah, I see! Looks like none of the other selectors in this file cast to `str`. What's the reason for that?

Would it be better to do:

```
if not context.activity.type:
    return False

if context.activity.type and isinstance(context.activity.type, ActivityTypes):
    return type == context.activity.type
else:
    return type == str(context.activity.type)
```

Reply:
just checked locally and it looks like `mypy` isn't throwing any type errors when the cast is removed, so we can update this to match the other selectors. Sorry for the confusion.